### PR TITLE
Pre-9.0.0 HF plugin init fixes.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
- Fix a problem with `updateSettings` not working properly when used to initialize the Formulas plugin. (#8056)
- Fix a problem when enabling some options caused the Formulas plugin to fail. (#8054)

Additionally, I added an argument for the `updatePlugin` method of the base plugin class.

### How has this been tested?
Added test cases for #8056.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8056 
2. #8054 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

[skip changelog]

### CODE REVIEW

**Review by nikersify — CHANGES_REQUESTED** [review]
```js
const hot = new Handsontable(el, {
	formulas: {
		engine: HyperFormula
	}
})

hot.updateSettings({formulas: false})

hot.updateSettings({formulas: {engine: HyperFormula}})
```

^ still throws `Error: There's no sheet with id = undefined`.

File: src/plugins/formulas/formulas.js [hunk 1]
```diff
@@ -177,7 +181,20 @@ export class Formulas extends BasePlugin {
       this.switchSheet(pluginSettings.sheetName);
     }
 
-    super.updatePlugin();
+    // If no data was passed to the `updateSettings` method and no sheet is connected to the instance -> create a
+    // new sheet using the currently used data. Otherwise, it will be handled by the `afterLoadData` call.
+    if (!newSettings.data && this.sheetName === null) {
```
[the remark this answers is not in the document]
Comment (jansiegel) [reply src/plugins/formulas/formulas.js:194]: This one's pretty tricky. 
When calling `updateSettings` with `data` present, it calls `loadData`, when calling it without `data` - it doesn't. 
Now - we have the sheet-creation and data-filling logic in the `afterLoadData` hook callback (it has to be there, because of some data-modifying logic in `loadData`). This creates problems with no sheet being present in this issue's scenario, when the data is initialized first, and formulas are being enabled using `updateSettings` (without `data` provided).
Basically, this condition ensures the sheet creation logic is not doubled (`updatePlugin` + `onAfterLoadData`).

**Comment by jansiegel** [comment]
> ```js
> const hot = new Handsontable(el, {
> 	formulas: {
> 		engine: HyperFormula
> 	}
> })
> 
> hot.updateSettings({formulas: false})
> 
> hot.updateSettings({formulas: {engine: HyperFormula}})
> ```
> 
> ^ still throws `Error: There's no sheet with id = undefined`.

Should be covered and tested with https://github.com/handsontable/handsontable/pull/8075/commits/62f62b4996c61fa9f85fa8f2e5654c87321145ba#diff-bae07df71dfecc0b0f016bb98971b965b624fc95c1e5e41ac8c9c0652ab18534R694.

File: src/plugins/formulas/formulas.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikersify) [reply src/plugins/formulas/formulas.js:194]: Thanks!

**Review by nikersify — APPROVED** [review]
Enabling the plugin with `enablePlugin` without applying any settings first it first, throws

```js
Uncaught TypeError: pluginSettings is undefined
    setupEngine register.js:17
    enablePlugin formulas.js:119
```

```js
const hot = new Handsontable(el, {
	data: [[1, 2, 3]]
})

hot.getPlugin('formulas').enablePlugin()
```

I checked the other plugins and it seems like `ContextMenu` also throws a weird error when enabling it like that with `enablePlugin()` (`TypeError: settings is undefined`). Not sure if we want to keep it like this because of the breaking change in the API this version, but also obviously won't block the merge because of this.

Other than that no comments from me.

**Review by nikersify — APPROVED** [review]
